internal/pkg/service/objectstorage/cloud/azure/storageaccount.go

```diff
@@ -0,0 +1,172 @@
+	}
+	return nil
+}
+
+func (s AzureObjectStorageAdapter) deleteStorageAccount(ctx context.Context, bucket *v1alpha1.Bucket, storageAccountName string) error {
+	// Delete Storage Account
+	// We delete the Storage Account, which delete the Storage Container
+	_, err := s.storageAccountClient.Delete(
+		ctx,
+		s.cluster.GetResourceGroup(),
+		storageAccountName,
+		nil,
+	)
+	if err != nil {
+		s.logger.Error(err, fmt.Sprintf("error deleting storage account %s and storage container %s", storageAccountName, bucket.Spec.Name))
+		return err
+	}
+	s.logger.Info(fmt.Sprintf("storage account %s and storage container %s deleted", storageAccountName, bucket.Spec.Name))
+	return nil
+}
+
+// existsStorageAccount checks if a storage account exists for the given bucket name in Azure Object Storage.
+// It returns a boolean indicating whether the storage account exists or not, along with any error encountered.
+func (s AzureObjectStorageAdapter) existsStorageAccount(ctx context.Context, storageAccountName string) (bool, error) {
+	availability, err := s.storageAccountClient.CheckNameAvailability(
+		ctx,
+		armstorage.AccountCheckNameAvailabilityParameters{
+			Name: to.Ptr(storageAccountName),
+			Type: to.Ptr("Microsoft.Storage/storageAccounts"),
+		},
+		nil)
+	if err != nil {
+		return false, err
+	}
+	return !*availability.NameAvailable, nil
+}
+
+// setLifecycleRules set a lifecycle rule on the Storage Account to delete Blobs older than X days
+func (s AzureObjectStorageAdapter) setLifecycleRules(ctx context.Context, bucket *v1alpha1.Bucket) error {
+	storageAccountName := sanitizeStorageAccountName(bucket.Spec.Name)
+	if bucket.Spec.ExpirationPolicy != nil {
+		_, err := s.managementPoliciesClient.CreateOrUpdate(
+			ctx,
+			s.cluster.GetResourceGroup(),
+			storageAccountName,
+			armstorage.ManagementPolicyNameDefault,
+			armstorage.ManagementPolicy{
+				Properties: &armstorage.ManagementPolicyProperties{
+					Policy: &armstorage.ManagementPolicySchema{
+						Rules: []*armstorage.ManagementPolicyRule{
+							{
+								Enabled: to.Ptr(true),
+								Name:    to.Ptr(LifecycleRuleName),
+								Type:    to.Ptr(armstorage.RuleTypeLifecycle),
+								Definition: &armstorage.ManagementPolicyDefinition{
+									Actions: &armstorage.ManagementPolicyAction{
+										BaseBlob: &armstorage.ManagementPolicyBaseBlob{
+											Delete: &armstorage.DateAfterModification{
+												DaysAfterModificationGreaterThan: to.Ptr[float32](float32(bucket.Spec.ExpirationPolicy.Days)),
+											},
+										},
+									},
+									Filters: &armstorage.ManagementPolicyFilter{
+										BlobTypes: []*string{
+											to.Ptr("blockBlob"),
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			nil,
+		)
+		if err != nil {
+			s.logger.Error(err, fmt.Sprintf("Error creating/updating Policy Rule for Storage Account %s", storageAccountName))
+		}
+		return err
```

Comment:
```suggestion
		return nil
```

Reply:
Well no, we log an error when there's an error but we still want ot return if (at least I think it was the intention

Reply:
Oh right, but this reads weird please do something like

```
		if err != nil {
			s.logger.Error(err, fmt.Sprintf("Error creating/updating Policy Rule for Storage Account %s", storageAccountName))
			return err
		}
		return nil
```

Reply:
done